src/overlays/actors/ovl_En_Mm/z_en_mm.c

```diff
@@ -44,39 +71,541 @@ static ColliderCylinderInit D_80AAEAE0 = {
+                    this->unk_254 &= ~1;
+                    EnMm_ChangeAnimation(this, RM_ANIM_HAPPY, &this->curAnimIndex);
+                } else {
+                    player->actor.textId = 0x202C;
+                    gSaveContext.infTable[23] |= 0x1000;
+                }
+                sp1C = 2;
+            }
+            break;
+        case 5:
+            if (func_80106BC8(globalCtx)) {
+                Player_UnsetMask(globalCtx);
+                Item_Give(globalCtx, ITEM_SOLD_OUT);
+                gSaveContext.itemGetInf[3] |= 0x800;
+                Rupees_ChangeBy(500);
+                player->actor.textId = 0x202E;
+                sp1C = 2;
+            }
+            break;
+        case 6:
+            if (func_80106BC8(globalCtx)) {
+                if ((player->actor.textId == 0x202E) || (player->actor.textId == 0x202C)) {
+                    this->unk_254 |= 1;
+                    EnMm_ChangeAnimation(this, RM_ANIM_SIT_WAIT, &this->curAnimIndex);
+                }
+                sp1C = 0;
+            }
+            break;
+    }
+
+    return sp1C;
+}
+
+s32 EnMm_GetTextId(EnMm* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    s32 textId;
+
+    textId = Text_GetFaceReaction(globalCtx, 0x1C);
+
+    if (gSaveContext.itemGetInf[3] & 0x800) {
+        if (textId == 0) {
+            textId = 0x204D;
+        }
+    } else if (player->currentMask == PLAYER_MASK_BUNNY) {
+        textId = (gSaveContext.infTable[23] & 0x1000) ? 0x202B : 0x202A;
+    } else if (textId == 0) {
+        textId = 0x2029;
+    }
+
+    return textId;
+}
+
+void func_80AADCD0(EnMm* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    f32 yawDiff;
+    s16 sp26;
+    s16 sp24;
+
+    if (this->unk_1E0 == 2) {
+        func_8010B720(globalCtx, player->actor.textId);
+        this->unk_1E0 = 1;
+    } else if (this->unk_1E0 == 1) {
+        this->unk_1E0 = func_80AADAA0(this, globalCtx);
+    } else {
+        if (func_8002F194(&this->actor, globalCtx)) {
+            this->unk_1E0 = 1;
+
+            if (this->curAnimIndex != 5) {
+                if ((this->actor.textId == 0x202A) || (this->actor.textId == 0x202B)) {
+                    EnMm_ChangeAnimation(this, RM_ANIM_EXCITED, &this->curAnimIndex);
+                    func_80078884(NA_SE_SY_TRE_BOX_APPEAR);
+                }
+            }
+        } else {
+            func_8002F374(globalCtx, &this->actor, &sp26, &sp24);
+            yawDiff = ABS((s16)(this->actor.yawTowardsPlayer - this->actor.shape.rot.y));
+
+            if ((sp26 >= 0) && (sp26 <= 0x140) && (sp24 >= 0) && (sp24 <= 0xF0) && (yawDiff <= 17152.0f) &&
+                (this->unk_1E0 != 3) && func_8002F2CC(&this->actor, globalCtx, 100.0f)) {
+                this->actor.textId = EnMm_GetTextId(this, globalCtx);
+            }
+        }
+    }
+}
+
+s32 EnMm_GetPointCount(Path* pathList, s32 pathNum) {
+    return (pathList + pathNum)->count;
```

Comment:
Can this not be matched with a normal array access?

Reply:
no it cant, and it happens in multiple places in the file, which tells me its real